GoogleForCreators/web-stories-wp
# Moveable: Update all elements together after group actions.

## Summary

Adds group actions of Moveable to the history as one entry.


## Relevant Technical Choices

Uses `updateElementsById` with an updater to update all the elements at the same time instead of using it inside loop.


## To-do

Integration test would be good in the future.


## User-facing changes

When the user resizes/drags/rotates a group and then clicks Undo/Redo, the Undo/Redo will be applied to all the elements together now instead of one by one.


## Testing Instructions

1. Add a few elements.
2. Multi-select.
3. Drag these elements around
4. Use Undo-Redo
5. Verify that Undo-Redo applies to all the elements at the same time in the multi-selection (as opposed to first Undoing the action for one element, then for the second, etc.)
6. Verify the same for Resizing.
7. Verify the same for Rotation.


---



Fixes #1657
